rideshare.rb

```diff
@@ -0,0 +1,57 @@
+rideshare_info = {
+  DR0001: [
+    {date: "3rd Feb 2016", cost: 10, rider: "RD0022", rating: 5},
+    {date: "3rd Feb 2016", cost: 30, rider: "RD0015", rating: 4},
+    {date: "5th Feb 2016", cost: 45, rider: "RD0003", rating: 2},
+  ],
+
+  DR0002: [
+    {date: "3rd Feb 2016", cost: 25, rider: "RD0073", rating: 5},
+    {date: "4th Feb 2016", cost: 15, rider: "RD0013", rating: 1},
+    {date: "5th Feb 2016", cost: 35, rider: "RD0066", rating: 3},
+  ],
+
+  DR0003: [
+    {date: "4th Feb 2016", cost: 5, rider: "RD0066", rating: 5},
+    {date: "5th Feb 2016", cost: 50, rider: "RD0003", rating: 2},
+  ],
+
+  DR0004: [
+    {date: "3rd Feb 2016", cost: 5, rider: "RD0022", rating: 5},
+    {date: "4th Feb 2016", cost: 10, rider: "RD0022", rating: 4},
+    {date: "5th Feb 2016", cost: 20, rider: "RD0073", rating: 5},
+  ],
+}
+
+rideshare_info.each do |driver, rides|
+  ride_count = rides.count
+  puts "#{driver} gave #{rides.count} rides"
+end
+
+def added_ride_info(mama_hash, key)
+  new_hash = {}
+  mama_hash.each do |driver, rides|
+    total = rides.sum { |ride| ride[key] }
```

Comment:
I love the idea of breaking this functionality out into a separate method, especially the idea of making it more abstract by passing in the field you're interested in summing. However, your names could be a little more descriptive. How about this:

```ruby
def sum_field_by_driver(rides_by_driver, key)
  driver_sums = {}
  rides_by_driver.each do |driver_id, rides|
    total = rides.sum { |ride| ride[key] }
    driver_sums[driver_id] = total
  end
  return driver_sums
end
```

Reply:
Thank you for the feedback! I agree that using more descriptive names improves the readability.